Don't disable the use of default libs, to prevent link failure on PowerPC.
https://sourceforge.net/p/silgraphite/bugs/56/

Use -Wimplicit-fallthrough only with clang, to prevent compile failure with gcc.
https://sourceforge.net/p/silgraphite/bugs/60/

Use -mfpmath=sse -msse2 only on Intel.
https://github.com/silnrsi/graphite/pull/44
--- src/CMakeLists.txt.orig	2018-08-15 00:38:09.000000000 -0500
+++ src/CMakeLists.txt	2018-10-01 12:46:50.000000000 -0500
@@ -136,9 +136,14 @@
 
 if  (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
     set_target_properties(graphite2 PROPERTIES
-        COMPILE_FLAGS   "-Wall -Wextra -Wno-unknown-pragmas -Wimplicit-fallthrough -Wendif-labels -Wshadow -Wno-ctor-dtor-privacy -Wno-non-virtual-dtor -fno-rtti -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden -mfpmath=sse -msse2"
-        LINK_FLAGS      "-nodefaultlibs"
+        COMPILE_FLAGS   "-Wall -Wextra -Wno-unknown-pragmas -Wendif-labels -Wshadow -Wno-ctor-dtor-privacy -Wno-non-virtual-dtor -fno-rtti -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden"
         LINKER_LANGUAGE C)
+    if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86|i.86")
+        add_definitions(-mfpmath=sse -msse2)
+    endif (${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86|i.86")
+    if (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
+        add_definitions(-Wimplicit-fallthrough)
+    endif (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     target_link_libraries(graphite2 c)
     include(Graphite)
     nolib_test(stdc++ $<TARGET_SONAME_FILE:graphite2>)
